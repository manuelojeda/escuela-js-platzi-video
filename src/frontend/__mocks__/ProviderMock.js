import React from 'react';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import initialState from '../initialState'
import { createBrowserHistory } from 'history';
import reducer from '../reducers';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)))
const history = createBrowserHistory();

const ProviderMock = (props) => (
  <Provider store={store}>
      <Router history={history}>
        {props.children}
      </Router>
  </Provider>
)

export default ProviderMock
