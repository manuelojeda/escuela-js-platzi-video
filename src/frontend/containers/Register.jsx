/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useState } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { registerRequest } from '../actions';
import '../assets/styles/components/Register.scss';
import HeaderLogin from '../components/HeaderLogin';
import Footer from '../components/Footer';

const Register = (props) => {
  const [form, setValues] = useState({
    name: '',
    email: '',
    password: '',
  });

  const handleInput = (event) => {
    setValues({
      ...form,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    props.registerRequest(form);
    props.history.push('/');
  };

  return (
    <div>
      <HeaderLogin />
      <section className='register'>
        <section className='register__container'>
          <h2>Regístrate</h2>
          <form className='register__container--form' onSubmit={handleSubmit}>
            <input
              className='input'
              type='text'
              placeholder='Nombre'
              name='name'
              onChange={handleInput}
            />
            <input
              className='input'
              type='text'
              placeholder='Correo'
              name='email'
              onChange={handleInput}
            />
            <input
              className='input'
              type='password'
              placeholder='Contraseña'
              name='password'
              onChange={handleInput}
            />
            <button className='button' type='submit'>Registrarme</button>
          </form>
          <Link to='/login'>
            Iniciar Sesión
          </Link>
        </section>
      </section>
      <Footer />
    </div>
  );
};

const mapDispatchToProps = {
  registerRequest,
};

export default connect(null, mapDispatchToProps)(Register);
