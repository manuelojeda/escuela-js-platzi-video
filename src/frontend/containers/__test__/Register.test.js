import React from 'react';
import { create } from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderMock from '../../__mocks__/ProviderMock';
import Register from '../Register';

configure({ adapter: new Adapter() });

describe('Header component', () => {
  test('Match Snapshot', () => {
    const header = create(
      <ProviderMock>
        <Register />
      </ProviderMock>,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
  it('Calls and executes preventDefault function onSubmit form', () => {
    const preventDefault = jest.fn();
    const wrapper = mount(
      <ProviderMock>
        <Register />
      </ProviderMock>,
    );
    wrapper.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    wrapper.unmount();
  });
});