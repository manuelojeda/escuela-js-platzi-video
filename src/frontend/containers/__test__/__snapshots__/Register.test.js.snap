// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Match Snapshot 1`] = `
<div>
  <header
    className="headerLogin"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Platzi Video"
        className="header__img"
        src="test-file-stubs"
      />
    </a>
  </header>
  <section
    className="register"
  >
    <section
      className="register__container"
    >
      <h2>
        Regístrate :v
      </h2>
      <form
        className="register__container--form"
        onSubmit={[Function]}
      >
        <input
          className="input"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          required={true}
          type="text"
        />
        <input
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="Correo"
          required={true}
          type="text"
        />
        <input
          className="input"
          name="password"
          onChange={[Function]}
          placeholder="Contraseña"
          required={true}
          type="password"
        />
        <button
          className="button"
          type="submit"
        >
          Registrarme
        </button>
      </form>
      <a
        href="/login"
        onClick={[Function]}
      >
        Iniciar Sesión
      </a>
    </section>
  </section>
  <footer
    className="footer"
  >
    <a
      href="/"
    >
      Terminos de uso
    </a>
    <a
      href="/"
    >
      Declaración de privacidad
    </a>
    <a
      href="/"
    >
      Centro de ayuda
    </a>
  </footer>
</div>
`;
